### PR TITLE
refactor(app): revert run a protocol from devices pages

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

allow users to start a run from robot overflow menus on the devices landing and device details pages. in part reverts the removal of the choose protocol slideout and devices landing overflow menu option in https://github.com/Opentrons/opentrons/pull/11660. however, instead of re-adding the "Run a protocol" button on the device details page, we are adding a "Run and protocol" overflow menu option to that page as well.

closes RCORE-327

# Changelog

 - Adds ability to run a protocol from the devices landing and device details robot overflow menus

# Review requests

run a protocol from each devices page

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import * as React from 'react'
+import { when } from 'jest-when'
+import { renderWithProviders } from '@opentrons/components'
+import { StaticRouter } from 'react-router-dom'
+import { fireEvent } from '@testing-library/react'
```
Comment (koji) [inline app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx:5]:
```suggestion
import { StaticRouter } from 'react-router-dom'
import { fireEvent } from '@testing-library/react'
import { renderWithProviders } from '@opentrons/components'
```

File: app/src/organisms/ChooseProtocolSlideout/index.tsx [hunk 2]
```diff
@@ -0,0 +1,519 @@
+import * as React from 'react'
+import path from 'path'
+import first from 'lodash/first'
+import { Trans, useTranslation } from 'react-i18next'
+import { Link, NavLink, useHistory } from 'react-router-dom'
+import { ApiHostProvider } from '@opentrons/react-api-client'
+import { useSelector } from 'react-redux'
+import { css } from 'styled-components'
```
Comment (koji) [inline app/src/organisms/ChooseProtocolSlideout/index.tsx:8]:
```suggestion
import * as React from 'react'
import path from 'path'
import first from 'lodash/first'
import { Trans, useTranslation } from 'react-i18next'
import { Link, NavLink, useHistory } from 'react-router-dom'
import { useSelector } from 'react-redux'
import { css } from 'styled-components'
import { ApiHostProvider } from '@opentrons/react-api-client'
```

File: app/src/organisms/ChooseProtocolSlideout/index.tsx [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+
+  const srcFileObjects =
+    selectedProtocol != null
+      ? selectedProtocol.srcFiles.map((srcFileBuffer, index) => {
+          const srcFilePath = selectedProtocol.srcFileNames[index]
+          return new File([srcFileBuffer], path.basename(srcFilePath))
+        })
+      : []
+
+  const [shouldApplyOffsets, setShouldApplyOffsets] = React.useState(true)
```
Comment (koji) [inline app/src/organisms/ChooseProtocolSlideout/index.tsx:75]:
```suggestion
  const [shouldApplyOffsets, setShouldApplyOffsets] = React.useState<boolean>(true)
```

File: app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      showSlideout: true,
+    })
+    expect(queryAllByText('Choose Protocol to Run')).not.toBeFalsy()
+    expect(queryAllByText(mockConnectableRobot.name)).not.toBeFalsy()
+  })
+  it('does not render slideout if showSlideout false', () => {
+    const [{ queryAllByText }] = render({
+      robot: mockConnectableRobot,
+      onCloseClick: jest.fn(),
+      showSlideout: true,
```
Comment (koji) [inline app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx:86]: from the test description, `showSlideout` should be `false`, shouldn't it?

File: app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    jest.resetAllMocks()
+  })
+
+  it('renders slideout if showSlideout true', () => {
+    const [{ queryAllByText }] = render({
+      robot: mockConnectableRobot,
+      onCloseClick: jest.fn(),
+      showSlideout: true,
+    })
+    expect(queryAllByText('Choose Protocol to Run')).not.toBeFalsy()
```
Comment (koji) [inline app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx:79]: `Choose Protocol to Run` is the header title, isn't it? 
Really need to use `queryAllByText`?

File: app/src/organisms/Devices/RobotOverflowMenu.tsx [hunk 6]
```diff
@@ -51,13 +75,40 @@ export function RobotOverflowMenu(props: RobotOverflowMenuProps): JSX.Element {
...
   let menuItems: React.ReactNode
-  if (robot.status === CONNECTABLE) {
+  if (robot.status === CONNECTABLE && runId == null) {
+    menuItems = (
+      <>
+        <MenuItem
+          {...targetProps}
+          onClick={handleClickRun}
+          disabled={isRobotOnWrongVersionOfSoftware}
+          data-testid={`RobotOverflowMenu_${robot.name}_runProtocol`}
```
Comment (koji) [inline app/src/organisms/Devices/RobotOverflowMenu.tsx:85]: This MenuItem uses `data-testid`, but the MenuItem for Robot Settings uses `id`. Is there any reason to do that?

**Review by koji — APPROVED** [review]
Tested this PR with a robot and it worked as expected.
Left a few comments and questions.

File: app/src/organisms/Devices/RobotOverflowMenu.tsx (hunk 6 above)
Comment (brenthagen) [reply app/src/organisms/Devices/RobotOverflowMenu.tsx:85]: this menu item is a reversion of https://github.com/Opentrons/opentrons/pull/11660/files#diff-1becb925beb535e61b25c04f59b5f918e94b81c4359e09b1ada7e2aa906014bd. the intention is to revert the removed functionality and testing to the previous state.

File: app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx (hunk 5 above)
Comment (brenthagen) [reply app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx:79]: most likely not - this is a reverted test file. to keep changes as atomic as possible i'll create a follow-up ticket to address test cases and code style for the reverted choose protocol slideout.

File: app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx (hunk 4 above)
Comment (brenthagen) [reply app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx:86]: see https://opentrons.atlassian.net/browse/RCORE-494

File: app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx (hunk 5 above)
Comment (brenthagen) [reply app/src/organisms/ChooseProtocolSlideout/__tests__/ChooseProtocolSlideout.test.tsx:79]: see https://opentrons.atlassian.net/browse/RCORE-494

**Pull request merged**